promql/parser/generated_parser.y

```diff
@@ -669,14 +669,14 @@ label_set_item  : IDENTIFIER EQL STRING
                         { $$ = labels.Label{Name: $1.Val, Value: yylex.(*parser).unquoteString($3.Val) } }
                 | string_identifier EQL STRING
                         { $$ = labels.Label{Name: $1.Val, Value: yylex.(*parser).unquoteString($3.Val) } }
+                | string_identifier
+                        { $$ = labels.Label{Name: labels.MetricName, Value: $1.Val} }
                 | IDENTIFIER EQL error
                         { yylex.(*parser).unexpected("label set", "string"); $$ = labels.Label{}}
                 | string_identifier EQL error
                         { yylex.(*parser).unexpected("label set", "string"); $$ = labels.Label{}}
                 | IDENTIFIER error
                         { yylex.(*parser).unexpected("label set", "\"=\""); $$ = labels.Label{}}
-                | string_identifier error
```

Comment:
It just moved several lines up -- string identifier without an operator is a metric name.  The go code will catch the case where the user tried to specify this multiple times, which is the remaining error case.

Reply:
This exact production did not move.

How do you know the full set of error cases?

Reply:
I'm not sure what you mean by "exact production"? 

The original case is saying that a string identifier cannot appear alone, without an operator (thus the error message, "expected ="). Since with the new syntax, we allow identifiers without an operator, it is no longer a parsing error to have a term without an operator. There are already tests in the parsing code to confirm that a metric can't have a name defined twice.

Are there other cases you are worried about?

Reply:
By "production" I mean a grammar rule. 

The rule I am talking about is `label_set_item : string_identifier error`.  
I would expect it to fire when anything unexpected appears after the identifier.
If you're confident it could not fire except in cases where the identifier appears alone, then that's fine.

Reply:
The containing rule, label_set_list, must be either `label_set_list COMMA label_set_item` or a bare `label_set_item`. That ensures that anything after the identifier close quote must be either `,` or `}`.

I added a couple of test cases, I think they cover what you're talking about.  (I see no problem with additional tests even if they may be redundant with other tests)